File: civitasjava/src/Civitas/SorpresaIrCasilla.java
package civitas;

import java.util.ArrayList;


public class SorpresaIrCasilla extends Casilla{
    private int valor;
    private Tablero tablero;

    public SorpresaIrCasilla(Tablero tablero, int valor, String texto){
        super(texto);
        this.valor = valor;
        this.tablero = tablero;
    }
    
    
    public void aplicarAJugador(int actual, ArrayList<Jugador> todos){
        if(jugadorCorrecto(actual, todos)){
            informe(actual, todos);
            int a = todos.get(actual).getNumCasillaActual();
            int tirada = tablero.calcularTirada(a,valor);
            int nuevaPos = tablero.nuevaPosicion(a, tirada);
            todos.get(actual).moverACasilla(nuevaPos);
            tablero.getCasilla(nuevaPos).recibeJugador(actual, todos);
        }
    }
    
    @Override
    public String toString(){
        return "SorpresaIrCasilla{ valor: " +valor +". tablero: " +tablero
                +super.toString() +"}";
    }
    
    
}

File: civitasruby/pkg/Civitas-0.0.1/lib/sorpresa_ir_casilla.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module Civitas
class SorpresaIrCasilla < Sorpresa
  def initialize(texto,tablero,valor)
    super(texto)
    @tablero = tablero
    @valor = valor
  end
  
  def aplicar_a_jugador(actual,todos)
      if(jugador_correcto(actual,todos))
        super(actual,todos)
        casilla_actual=todos[actual].numCasillaActual
        tirada=@tablero.calcular_tirada(casilla_actual,@valor)
        nueva_pos=@tablero.nueva_posicion(casilla_actual,tirada)
        todos[actual].mover_a_casilla(nueva_pos)
        @tablero.get_casilla(nueva_pos).recibe_jugador(actual, todos)
      end
  end
  
  def to_string
    a = "Ir Casilla"
    return a
  end
  public_class_method :new
end
end